### PR TITLE
all: rename module and move `main` pacakge to `cmd/asdf`

### LINKED ISSUE #84: proposal: move `main` package into `cmd/asdf` and rename module for compatibility with `go install`
### Is your feature request related to a problem? Please describe

Installing the new Go version is a bit strangely roundabout, especially for a Go CLI tool.

### Describe the proposed solution

The convention for Go projects is to put `main` packages in a subdirectory of `cmd` named the same as what the output binary should be so that they can be easily installed with, for example, `go install github.com/asdf-vm/asdf-core-go/cmd/asdf@latest`. This would also make it possible to keep ASDF up to date via, for example, [gup](https://github.com/nao1215/gup) as a replacement for the removed `asdf update` command.

The module name will also need to be updated to follow Go conventions, changing from `asdf` to `github.com/asdf-vm/asdf-core-go`.

### Describe similar `asdf` features and why they are not sufficient

`asdf update` is being removed in the Go rewrite.

### Describe other workarounds you've considered

N/A.

### PR DESCRIPTION
**Pull request by DeedleFake**
Fixes #84.

This also fixes a few other minor things, but in so doing causes `go test` to fail because a test was not being run at all before due to being named incorrectly is now running and it fails.

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -12,19 +12,18 @@ import (
...
-	"asdf/internal/help"
-	"asdf/internal/info"
-	"asdf/internal/installs"
-	"asdf/internal/plugins"
-	"asdf/internal/resolve"
-	"asdf/internal/shims"
-	"asdf/internal/toolversions"
-	"asdf/internal/versions"
-
+	"github.com/asdf-vm/asdf-core-go/internal/config"
```
Comment (Stratus3D) [inline cli/cli.go:15]: This repo (github.com/asdf-vm/asdf-core-go) is temporary and will be archived in a couple weeks. All the changes in this repo will soon be merged into github.com/asdf-vm/asdf once most of the test code in `main_test.go` is passing.

Should we make this name `github.com/asdf-vm/asdf`? What's the difference between `github.com/asdf-vm/asdf` and `asdf`?

**Comment by Stratus3D** [comment]
Thanks for the PR @DeedleFake. I left a few comments. Looks like PR checks failed, I will investigate at a later time. Otherwise looks like something I can merge.

File: cli/cli.go (hunk 1 above)
Comment (DeedleFake) [reply cli/cli.go:15]: The module name should match the import path that identifies the repo. If the plan is to move the code into another repo, then it should match the repo it's moved to. It can be renamed now or when it's moved. Alternatively, you can use [a vanity URL](https://pkg.go.dev/cmd/go#:~:text=If%20the%20import,proxy%20protocol.), i.e. `asdf-vm.com`. Then people could, for example, use `go install asdf-vm.com/cmd/asdf@latest` to install the command. That would allow you to move repos without having to rename the module. I use a vanity URL, `deedles.dev`, for my personal projects.

Comment (Stratus3D) [reply cli/cli.go:15]: Let's go with `github.com/asdf-vm/asdf` since that is where this code will live in a few weeks. I don't expect many people to use this `asdf-core-go` repo, and if they do they'll hopefully see it's not ready for regular use. I've updated the repo description to make this clear.

**Comment by DeedleFake** [comment]
I finally got a second to go through some of the code and see if there was anything else I could make a bit more idiomatic or otherwise clean up. Just out of curiosity, is there any particular reason that you're using Go 1.21 instead of 1.23? There are a couple of places where it would be nice to be able to use `iter.Seq`s. For example, the `readLines()` function in `internal/toolversions/toolversions.go` allocates several intermediary slices in a row that would be unnecessary with iterators, although [one of those will have to wait until Go 1.24 to address](https://pkg.go.dev/strings@master#SplitSeq).

**Comment by Stratus3D** [comment]
> First of all, is there any particular reason that you're using Go 1.21 instead of 1.23?

No, I think 1.21 was the latest version available when I started? I will upgrade today. I guess I'm happy not using `iter` right now.

> Second of all, parseLine() in internal/toolversions is essentially just replicating strings.Fields() except that it requires at least one space specifically, not just any whitespace, between the fields. Can I just replace that? I figured that I should ask because that technically would change the functionality slightly in that tabs and a few other whitespace characters would become legal for separating the parts of a .tool-versions line.

Yes, I think it'd be fine to permit tabs and other whitespace characters as separators.

File: internal/toolversions/toolversions_test.go [hunk 2]
```diff
@@ -133,7 +133,7 @@ func TestgetAllToolsAndVersionsInContent(t *testing.T) {
 		{
 			desc:  "returns empty list with found true and no error when empty content",
 			input: "",
-			want:  []ToolVersions{},
+			want:  []ToolVersions(nil),
```
Comment (Stratus3D) [inline internal/toolversions/toolversions_test.go:136]: What's the difference between old and new?

**Comment by Stratus3D** [comment]
Hi @DeedleFake I fixed up the GitHub workflows so your changes here pass and merged them to master in https://github.com/asdf-vm/asdf-core-go/pull/92

**Comment by Stratus3D** [comment]
@DeedleFake the Go version has been updated on `master`.

File: internal/toolversions/toolversions_test.go (hunk 2 above)
Comment (DeedleFake) [reply internal/toolversions/toolversions_test.go:136]: The original version of the test was failing because the function itself returned `nil` instead of an empty slice. Most code treats `nil` slices and empty slices as being equivalent, although there are a few exceptions. It's generally easy to do because the only significant way in which nil slices and empty slices behave differently, ignoring things like JSON encoding, is that `emptySlice != nil`.

After changing the test's expected output, however, I also went and changed the test itself to better handle the case of empty slices, so now this change doesn't really make any difference.

**Comment by DeedleFake** [comment]
Ah, whoops. Somehow I completely missed that you had merged them in that previous comment. Sorry about that.
